### PR TITLE
[IOTDB-3025] add permission check

### PR DESCRIPTION
**Pull request by RYH61**
1、add permission check
2、add tsblock

### CODE REVIEW

File: server/src/main/java/org/apache/iotdb/db/service/thrift/impl/DataNodeTSIServiceImpl.java [hunk 1]
```diff
@@ -270,6 +270,11 @@ public TSExecuteStatementResp executeStatement(TSExecuteStatementReq req) {
         StatementGenerator.createStatement(
             statement, SESSION_MANAGER.getZoneId(req.getSessionId()));
 
+    TSStatus status = AuthorizerManager.getInstance().checkAuthority(s, req.sessionId);
+    if (status.getCode() != TSStatusCode.SUCCESS_STATUS.getStatusCode()) {
+      return RpcUtils.getTSExecuteStatementResp(status);
+    }
+
```
Comment (MarcosZyk) [inline server/src/main/java/org/apache/iotdb/db/service/thrift/impl/DataNodeTSIServiceImpl.java:277]: The analysis step has not been processed and the pathPattern in statement has not been parsed and the wildcard has not been removed.

Comment (RYH61) [reply server/src/main/java/org/apache/iotdb/db/service/thrift/impl/DataNodeTSIServiceImpl.java:277]: ok i will change it next

File: server/src/main/java/org/apache/iotdb/db/mpp/sql/statement/crud/InsertRowStatement.java [hunk 2]
```diff
@@ -57,6 +58,11 @@ public InsertRowStatement() {
     statementType = StatementType.INSERT;
   }
 
+  @Override
+  public List<? extends PartialPath> getPaths() {
+    return null;
```
[the remark this answers is not in the document]
Comment (RYH61) [reply server/src/main/java/org/apache/iotdb/db/mpp/sql/statement/crud/InsertRowStatement.java:63]: Yes, we need to return the specified path, here I will modify it again

File: thrift-confignode/src/main/thrift/confignode.thrift [hunk 3]
```diff
@@ -143,6 +143,12 @@ struct TLoginReq {
     2: required string password
 }
 
+struct TcheckUserPrivilegesReq{
```
[the remark this answers is not in the document]
Comment (RYH61) [reply thrift-confignode/src/main/thrift/confignode.thrift]: fix

**Review by MarcosZyk — APPROVED** [review]

File: confignode/src/main/java/org/apache/iotdb/confignode/manager/ConfigManager.java [hunk 4]
```diff
@@ -367,4 +367,14 @@ public TSStatus login(String username, String password) {
       return status;
     }
   }
+
+  @Override
+  public TSStatus checkUserPrivileges(String username, List<String> paths, int permission) {
+    TSStatus status = confirmLeader();
+    if (status.getCode() == TSStatusCode.SUCCESS_STATUS.getStatusCode()) {
+      return AuthorInfo.getInstance().checkUserPrivileges(username, paths, permission);
```
Comment (wangchao316) [inline confignode/src/main/java/org/apache/iotdb/confignode/manager/ConfigManager.java:375]: please use PermissionManager.checkUserPrivileges

File: confignode/src/main/java/org/apache/iotdb/confignode/persistence/AuthorInfo.java [hunk 5]
```diff
@@ -77,7 +78,37 @@ public TSStatus login(String username, String password) {
     return tsStatus;
   }
 
-  public TSStatus authorNonQuery(AuthorReq authorReq) throws AuthException {
+  public TSStatus checkUserPrivileges(String username, List<String> paths, int permission) {
```
Comment (wangchao316) [inline confignode/src/main/java/org/apache/iotdb/confignode/persistence/AuthorInfo.java:81]: remove to PermissionManager,  PermissionManager call AuthorInfo

File: thrift-confignode/src/main/thrift/confignode.thrift [hunk 6]
```diff
@@ -188,4 +194,6 @@ service ConfigIService {
   TAuthorizerResp queryPermission(TAuthorizerReq req)
 
   common.TSStatus login(TLoginReq req)
+
+  common.TSStatus checkUserPrivileges(TCheckUserPrivilegesReq req)
```
Comment (wangchao316) [inline thrift-confignode/src/main/thrift/confignode.thrift:198]: add some UT for checkUserPrivileges

**Review by wangchao316 — CHANGES_REQUESTED** [review]

File: confignode/src/main/java/org/apache/iotdb/confignode/persistence/AuthorInfo.java (hunk 5 above)
Comment (RYH61) [reply confignode/src/main/java/org/apache/iotdb/confignode/persistence/AuthorInfo.java:81]: fix

File: thrift-confignode/src/main/thrift/confignode.thrift (hunk 6 above)
Comment (RYH61) [reply thrift-confignode/src/main/thrift/confignode.thrift:198]: fix

**Pull request merged**